pkg/networkservice/common/mechanisms/recvfd/server_test.go

```diff
@@ -220,3 +224,76 @@ func (s *checkRecvfdTestSuite) TestRecvfdClosesMultipleFiles() {
 		}, time.Second, time.Millisecond*100)
 	}
 }
+
+func TestRecvfdDoesntWaitForAnyFilesOnRequestsFromBegin(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+
+	t.Cleanup(func() {
+		cancel()
+		goleak.VerifyNone(t)
+	})
+
+	// Create a server
+	var eventFactory begin.EventFactory
+	server := chain.NewNetworkServiceServer(
+		begin.NewServer(),
+		checkcontextonreturn.NewServer(t, func(t *testing.T, ctx context.Context) {
+			eventFactory = begin.FromContext(ctx)
```

Comment:
should be used synchronization to fix race cond between test goroutine and request goroutine